packages/react-components/src/shared/src/resolveChildren.ts

```diff
@@ -0,0 +1,19 @@
+import { Fragment, ReactElement, ReactNode } from "react";
+import { isFunction, isNil } from "lodash";
+
+// Support first level fragment: https://github.com/facebook/react/issues/11859.
+export function resolveFragment(children: ReactNode) {
+    if (!isNil(children) && (children as ReactElement).type === Fragment) {
+        return (children as ReactElement).props.children;
+    }
+
+    return children;
+}
+
+export function resolveChildren<T extends any[]>(children: ReactNode | ((args: T[]) => ReactNode), renderProps?: T): ReactNode {
```

Comment:
Could we make custom types for repetive React patterns? For example ReactChildren would make sense for `ReactNode | ((args: T[]) => ReactNode)`

Reply:
Might be hard though since you can't really know which args the function signature could have

Reply:
Not sure about the generic constraint T.

From what I understand the constraint is to ensure T is an array. How come then you still have to specify T as an array? `((args: T[])`